File: week8/partialAbstract.cpp
#include<iostream>

using namespace std;

class Shapes
{
    private:
    virtual float area()=0;
    public:
    void displayMsg()
    {
        cout<<"These are the shapes\n";
    }
};
class Triangle: public Shapes
{
    public:
    float b,h;
    Triangle(float base,float height)
    {
        b=base;
        h=height;
    }
    float area()
    {
        return 0.5*b*h;
    }
};
class Circle: public Shapes
{
    public:
    float r;
    Circle(float radius)
    {
        r=radius;
    }
    float area()
    {
        return 3.14*r*r;
    }
};
int main()
{
    Triangle tri(6,4);
    Circle cir(8.5);
    tri.displayMsg();
    cout<<"The area of a Triangle : "<<tri.area()<<endl;
    cir.displayMsg();
    cout<<"The area of a Circle: "<<cir.area()<<endl;
    return 0;
}

File: week8/partialAbstract.java
abstract class Shape
{
    abstract float area();
    void display()
    {
        System.out.println("These are the shapes");
    }
}
class Rectangle extends Shape
{
    float a,b;
    Rectangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return a*b;
    }
}
class Triangle extends Shape
{
    float a,b;
    Triangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return (float)0.50*a*b;
    }
}
public class Main
{
    public static void main(String[] args)
    {
        Rectangle obj=new Rectangle(16,25);
        Triangle obj1=new Triangle(10,6);
        System.out.println("The area of a Rectangle : "+obj.area());
        System.out.println("The area of a Triangle : "+obj1.area());
    }
}
